wp/lib/bap_wp/src/precondition.ml

```diff
@@ -1302,9 +1309,9 @@ let mem_read_offsets (env2 : Env.t) (offset : Constr.z3_expr -> Constr.z3_expr)
   if List.is_empty conds then
     None
   else
-    Some (Assume (AfterExec (Constr.mk_goal name (Bool.mk_and ctx conds))))
+    Some (Assume (AfterExec (Constr.mk_goal name (Z3_utils.mk_and ctx conds))))
 
-let check ?refute:(refute = true) ?(print_constr = []) ?(debug = false)
+let check ?refute:(refute = true) ?(print_constr = []) ?(debug = false) ?ext_solver
```

Comment:
Can we add the default value for `ext_solver` to be more consistent with the other arguments in this function?

Reply:
We can, but the default value is None. Is that what you mean?

Reply:
Yup! The optional arguments are all written as `?(arg = default)`, so I think changing to `?(refute = true)` and `?(ext_solver = None)` would be more consistent.

Reply:
Hmm. Actually I think I'm using a slightly different optional keyword argument mechanism here that for the other arguments. It doesn't accept an explicit (ext_solver = None).